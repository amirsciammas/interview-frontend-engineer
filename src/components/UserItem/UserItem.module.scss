.UserItem {
  padding: 30px;
  border-radius: 6px;
  background-color: white;
  box-shadow: 0px 3px 5px -1px rgba(0, 0, 0, 0.2),
    0px 6px 10px rgba(0, 0, 0, 0.14), 0px 1px 18px rgba(0, 0, 0, 0.12);
  cursor: pointer;
}

.UserItem:hover {
  outline: 2px solid grey;
}

.UserItem:active {
  transform: scale(0.98);
  box-shadow: 3px 2px 22px 1px rgb(0 0 0 / 24%);
}

.name {
  font-weight: bold;
  align-self: center;
}
