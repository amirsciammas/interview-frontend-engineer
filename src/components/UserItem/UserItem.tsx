import React, { FC } from "react";
import { User } from "../../interfaces/UserType";
import styles from "./UserItem.module.scss";
import Grid from "@mui/material/Grid";

interface UserItemProps {
  user: User;
}

const UserItem: FC<UserItemProps> = ({ user }) => (
  <div className={styles.userItem} data-testid="UserItem">
    <Grid container>
      <Grid item xs={6} className={styles.name}>
        <div>{user.name}</div>
        <div>"{user.username}"</div>
      </Grid>
      <Grid item xs={6}>
        <div>{user.email}</div>
        <div>{user.website}</div>
      </Grid>
    </Grid>
  </div>
);

export default UserItem;
