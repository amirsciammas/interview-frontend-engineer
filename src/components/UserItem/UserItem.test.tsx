import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import UserItem from "./UserItem";
import renderer from 'react-test-renderer';

const user = {
  id: 0,
  name: "John",
  username: "JDoe",
  email: "john.doe@gmail.com",
  address: {
    street: "",
    suite: "",
    city: "",
    zipcode: "",
    geo: {
      lat: "",
      lng: "",
    },
  },
  phone: "",
  website: "",
  company: {
    name: "",
    catchPhrase: "",
    bs: "",
  },
};

describe("<UserItem />", () => {
  test("it should mount", () => {
    render(<UserItem user={user} />);
    const userItem = screen.getByTestId("UserItem");
    expect(userItem).toBeInTheDocument();
  });
  test("renders user data", () => {
    render(<UserItem user={user} />);
    const userItemName = screen.getByTestId("UserItem.name");
    expect(userItemName.textContent).toBe(user.name);

    const userItemUsername = screen.getByTestId("UserItem.username");
    expect(userItemUsername.textContent).toBe(`"${user.username}"`);

    const userItemEmail = screen.getByTestId("UserItem.email");
    expect(userItemEmail.textContent).toBe(user.email);

    const userItemWebsite = screen.getByTestId("UserItem.website");
    expect(userItemWebsite.textContent).toBe(user.website);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<UserItem user={user} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
