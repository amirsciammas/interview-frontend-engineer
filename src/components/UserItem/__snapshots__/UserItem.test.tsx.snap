// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserItem /> renders correctly 1`] = `
<div
  className="userItem"
  data-testid="UserItem"
>
  <div
    className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 name css-1osj8n2-MuiGrid-root"
    >
      <div
        data-testid="UserItem.name"
      >
        John
      </div>
      <div
        data-testid="UserItem.username"
      >
        "
        JDoe
        "
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
    >
      <div
        data-testid="UserItem.email"
      >
        john.doe@gmail.com
      </div>
      <div
        data-testid="UserItem.website"
      >
        
      </div>
    </div>
  </div>
</div>
`;
